### ISSUE TITLE
All kubevirt component recreate after operator spec change

### ISSUE THREAD

**Issue by AllenXu93** [kubevirt/kubevirt#13209]
**What happened**:
I want to update virt-controller, so update virt-operator's  env `VIRT_CONTROLLER_IMAGE` to a new image url.
After I update operator, all the kubevirt's component will be updated, these component's pods will recreated

**What you expected to happen**:
Only virt-controller deploy update.

**How to reproduce it (as minimally and precisely as possible)**:
Edit virt-operator's env, like `VIRT_CONTROLLER_IMAGE`

**Additional context**:
I compare the rs of one component like `virt-api`, 

```
kubectl -n kubevirt get rs virt-api-58c585c8bb -oyaml > /tmp/virt-api-58c585c8bb.yaml
kubectl -n kubevirt get rs virt-api-865f698f9f -oyaml > /tmp/virt-api-865f698f9f.yaml
diff /tmp/virt-api-865f698f9f.yaml /tmp/virt-api-58c585c8bb.yaml
```

result:
```
diff /tmp/virt-api-865f698f9f.yaml /tmp/virt-api-58c585c8bb.yaml
7c7
<     deployment.kubernetes.io/revision: "3"
---
>     deployment.kubernetes.io/revision: "4"
10c10
<     kubevirt.io/install-strategy-identifier: 99f476c2738c8d6576a0775e6c62ec1f27d3439d
---
>     kubevirt.io/install-strategy-identifier: 67216920ac2ee521eabaa18d48a0b276f2555936
13,14c13,14
<   creationTimestamp: "2024-09-12T02:33:56Z"
<   generation: 3
---
>   creationTimestamp: "2024-11-08T06:53:34Z"
>   generation: 2
20c20
<     pod-template-hash: 865f698f9f
---
>     pod-template-hash: 58c585c8bb
22c22
<   name: virt-api-865f698f9f
---
>   name: virt-api-58c585c8bb
31,32c31,32
<   resourceVersion: "74521312"
<   uid: addc03e0-9795-4fde-a140-0c1fabb558a5
---
>   resourceVersion: "74521783"
>   uid: 0dab5f13-97ad-4572-b993-82397701dd05
34c34
<   replicas: 1
---
>   replicas: 2
38c38
<       pod-template-hash: 865f698f9f
---
>       pod-template-hash: 58c585c8bb
42c42
<         kubevirt.io/install-strategy-identifier: 99f476c2738c8d6576a0775e6c62ec1f27d3439d
---
>         kubevirt.io/install-strategy-identifier: 67216920ac2ee521eabaa18d48a0b276f2555936
51c51
<         pod-template-hash: 865f698f9f
---
>         pod-template-hash: 58c585c8bb
150,154c150,154
<   availableReplicas: 1
<   fullyLabeledReplicas: 1
<   observedGeneration: 3
<   readyReplicas: 1
<   replicas: 1
---
>   availableReplicas: 2
>   fullyLabeledReplicas: 2
>   observedGeneration: 2
>   readyReplicas: 2
>   replicas: 2
```

The main different is operator update a pod label: `kubevirt.io/install-strategy-identifier`,  `virt-operator` deployment spec change will cause this label change, which cause all deployment updated.

**Environment**:
- KubeVirt version (use `virtctl version`): v1.3.1
- Kubernetes version (use `kubectl version`): v1.28.11

**Comment by xpivarc** [comment]
I tend to lean that this is expected behavior. Is this causing any production issues?

**Comment by AllenXu93** [comment]
> I tend to lean that this is expected behavior. Is this causing any production issues?

For example if cluster has many nodes,  the whole update will be very slow because of virt-handler daemonset rollingupdate, and I'm not sure whether virt-handler's update will affect vm which is in creation or migration.
I think component which do not have any change should not be updated , is there any considerations that update all kubevirt's component when operator spec chagne ?

**Comment by iholder101** [comment]
@xpivarc Do you think it is necessary to re-deploy all pods in this case? If so, what's the reason behind it?

Intuitively I think the problem reported in this issue makes sense.
However, there's a lot of mess around virt-operator, which needs to be heavily refactored (if not completely re-written). Saying that, I'm not sure this is a high priority issue for the project. @AllenXu93 do you think you'll have the time to take this upon yourself?

**Comment by AllenXu93** [comment]
> @xpivarc Do you think it is necessary to re-deploy all pods in this case? If so, what's the reason behind it?
> 
> Intuitively I think the problem reported in this issue makes sense. However, there's a lot of mess around virt-operator, which needs to be heavily refactored (if not completely re-written). Saying that, I'm not sure this is a high priority issue for the project. @AllenXu93 do you think you'll have the time to take this upon yourself?

Yes, I would like to contribute

**Comment by vladikr** [comment]
> of virt-handler daemonset rollingupdate, and I'm not sure whether virt-handler's update will affect vm which is in creation or migration.

It probably will, but both migration and the creation can be re-tried

Normally, such updates are taking place during system updates.
By not redeploying, we risk kubevirt components being out of sync regarding version compatibility.

**Comment by iholder101** [comment]
> > of virt-handler daemonset rollingupdate, and I'm not sure whether virt-handler's update will affect vm which is in creation or migration.
> 
> It probably will, but both migration and the creation can be re-tried
> 
> Normally, such updates are taking place during system updates. By not redeploying, we risk kubevirt components being out of sync regarding version compatibility.

I agree in the case of system upgrades, since there we'll probably replace all image components anyhow.
However, does this behavior make sense when simply replacing one of our components' image?
